packages/manager/modules/netapp/src/dashboard/volumes/dashboard/routing.js

```diff
@@ -0,0 +1,65 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('netapp.dashboard.volumes.dashboard', {
+    url: '/:volumeId',
+    views: {
+      'netappContainer@netapp': {
+        component: 'ovhManagerNetAppVolumesDashboard',
+      },
+    },
+    resolve: {
+      goToVolumeDashboard: /* @ngInject */ (
+        $state,
+        Alerter,
+        serviceName,
+        volumeId,
+      ) => (message = false, type = 'success') => {
+        const reload = message && type === 'success';
+        const promise = $state.go(
+          'netapp.dashboard.volumes.dashboard',
+          {
+            serviceName,
+            volumeId,
+          },
+          {
+            reload,
+          },
+        );
+
+        if (message) {
+          promise.then(() => {
+            Alerter[type](message);
+          });
+        }
+
+        return promise;
+      },
+      volumeId: /* @ngInject */ ($transition$) =>
+        $transition$.params().volumeId,
+      volume: /* @ngInject */ (volumes, volumeId) =>
+        volumes.find(({ id }) => id === volumeId),
+      volumeDashboardLink: /* @ngInject */ ($state, $transition$) =>
+        $state.href(
+          'netapp.dashboard.volumes.dashboard',
+          $transition$.params(),
+        ),
+      accessPath: /* @ngInject */ ($http, serviceName, volumeId) =>
+        $http
+          .get(`/storage/netapp/${serviceName}/share/${volumeId}/accessPath`)
+          .then(({ data }) => data)
+          .then((accessPaths) =>
+            accessPaths.find(({ preferred }) => !!preferred),
+          )
+          .catch(() => null),
+      updateVolume: /* @ngInject */ ($http, $translate, serviceName) => ({
+        name,
+        id,
+        description,
+      }) =>
+        $http.put(`/storage/netapp/${serviceName}/share/${id}`, {
```

Comment:
any plans to move these to a service?

Reply:
I don't see any benefits here as call is used once, can only update name/description for now and this will only add more code